tests/integration/graviton_test.py

```diff
@@ -908,6 +908,65 @@ def while_continue_accumulation(n):
         )
 
 
+def blackbox_pyteal_named_tupleness_test():
+    from typing import Literal as L
+    from tests.blackbox import Blackbox
+    from pyteal import (
+        Seq,
+        abi,
+        Subroutine,
+        TealType,
+        Return,
+        And,
+        Mode,
+    )
+
+    class NamedTupleExample(abi.NamedTuple):
+        a: abi.Bool
+        b: abi.Address
+        c: abi.Tuple2[abi.Uint64, abi.Bool]
+        d: abi.StaticArray[abi.Byte, L[10]]
+        e: abi.StaticArray[abi.Bool, L[4]]
+        f: abi.Uint64
+
+    sdk_abi_t = abi.algosdk_from_annotation(NamedTupleExample)
+
+    @Blackbox(input_types=[None])
+    @Subroutine(TealType.uint64)
+    def named_tuple_field_access(v_temp: NamedTupleExample):
+        return Seq(
+            (v_a0 := abi.Bool()).set(v_temp[0]),
+            (v_a1 := abi.Bool()).set(v_temp.a),
+            (v_b0 := abi.Address()).set(v_temp[1]),
+            (v_b1 := abi.Address()).set(v_temp.b),
+            (v_c0 := abi.make(abi.Tuple2[abi.Uint64, abi.Bool])).set(v_temp[2]),
+            (v_c1 := abi.make(abi.Tuple2[abi.Uint64, abi.Bool])).set(v_temp.c),
+            (v_d0 := abi.make(abi.StaticArray[abi.Byte, L[10]])).set(v_temp[3]),
+            (v_d1 := abi.make(abi.StaticArray[abi.Byte, L[10]])).set(v_temp.d),
+            (v_e0 := abi.make(abi.StaticArray[abi.Bool, L[4]])).set(v_temp[4]),
+            (v_e1 := abi.make(abi.StaticArray[abi.Bool, L[4]])).set(v_temp.e),
+            (v_f0 := abi.Uint64()).set(v_temp[5]),
+            (v_f1 := abi.Uint64()).set(v_temp.f),
+            Return(
+                And(
+                    v_a0.get() == v_a1.get(),
+                    v_b0.get() == v_b1.get(),
+                    v_c0.encode() == v_c1.encode(),
+                    v_d0.encode() == v_d1.encode(),
+                    v_e0.encode() == v_e1.encode(),
+                    v_f0.get() == v_f1.get(),
+                )
+            ),
+        )
```

Comment:
Could you add one where we set the values on the NamedTuple?

Reply:
yes, it now has both setting and getting (under name not index).